dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -307,7 +307,8 @@ private ListenableFuture<Response> handleThrowable(Throwable clientSideThrowable
 
         private ListenableFuture<Response> incrementFailuresAndMaybeRetry(
                 Response response, BiFunction<Endpoint, Response, Throwable> failureSupplier, Meter meter) {
-            if (++failures <= maxRetries) {
+            failures = failures + 1 + Responses.getProxyRetryAttempts(response);
```

Comment:
note that this retains the `+1` from before in addition to the retry attempts value. I'm not certain if we'd implement this as number of retry attempts (total attempts - 1) or report the total attempts.

I suppose if we're reporting total attempts, perhaps `Proxy-Request-Attempts` may be a more accurate name than `Proxy-Retry-Attempts`, but either model will fit, I'll just need to adjust the floor by 1.

Reply:
Fwiw the language I usually use is:
- if 5 total requests are sent:
  - the first one is an attempt, and the latter 4 are "retries" but also attempts
  - so that's "5 attempts" and "4 retries"
- if 1 total request is sent, that's 1 attempt and 0 retries

Reply:
Yes, it depends where we land on naming the new header :-)